homeassistant/components/nasweb/climate.py

```diff
@@ -0,0 +1,168 @@
+            identifiers={(DOMAIN, self._thermostat.webio_serial)}
+        )
+
+    async def async_added_to_hass(self) -> None:
+        """When entity is added to hass."""
+        await super().async_added_to_hass()
+        self._handle_coordinator_update()
+
+    def _set_attr_available(
+        self, entity_last_update: float, available: bool | None
+    ) -> None:
+        if (
+            self.coordinator.last_update is None
+            or time.time() - entity_last_update >= STATUS_UPDATE_MAX_TIME_INTERVAL
+        ):
+            self._attr_available = False
+        else:
+            self._attr_available = available if available is not None else False
+
+    @callback
+    def _handle_coordinator_update(self) -> None:
+        """Handle updated data from the coordinator."""
+        self._attr_current_temperature = self._thermostat.current_temp
+        self._attr_target_temperature_low = self._thermostat.temp_target_min
+        self._attr_target_temperature_high = self._thermostat.temp_target_max
+        self._attr_hvac_mode = self._get_current_hvac_mode()
+        self._attr_hvac_action = self._get_current_action()
+        self._attr_name = self._thermostat.name if self._thermostat.name else None
+        self._set_attr_available(
+            self._thermostat.last_update, self._thermostat.available
+        )
+        self.async_write_ha_state()
+
+    def _get_current_hvac_mode(self) -> HVACMode:
+        have_cooling = self._thermostat.enabled_above_output
+        have_heating = self._thermostat.enabled_below_output
+        if have_cooling and have_heating:
+            return HVACMode.HEAT_COOL
+        if have_cooling:
+            return HVACMode.COOL
+        if have_heating:
+            return HVACMode.HEAT
+        if self._thermostat.enabled_inrange_output:
+            return HVACMode.FAN_ONLY
+        return HVACMode.OFF
+
+    def _get_current_action(self) -> HVACAction:
+        if self._thermostat.current_temp is None:
+            return HVACAction.OFF
+        if (
+            self._thermostat.temp_target_min is not None
+            and self._thermostat.current_temp < self._thermostat.temp_target_min
+            and self._thermostat.enabled_below_output
+        ):
+            return HVACAction.HEATING
+        if (
+            self._thermostat.temp_target_max is not None
+            and self._thermostat.current_temp > self._thermostat.temp_target_max
+            and self._thermostat.enabled_above_output
+        ):
+            return HVACAction.COOLING
+        if (
+            self._thermostat.temp_target_min is not None
+            and self._thermostat.temp_target_max is not None
+            and self._thermostat.current_temp > self._thermostat.temp_target_min
+            and self._thermostat.current_temp < self._thermostat.temp_target_max
+            and self._thermostat.enabled_inrange_output
+        ):
+            return HVACAction.FAN
+        return HVACAction.IDLE
```

Comment:
The logic here will claim current action is `HVACAction.IDLE` if `current_temp == temp_target_min`  or `current_temp == temp_target_max`.
That doesn't seem correct?